src/lib/pmd/OutputProcessor.ts

```diff
@@ -0,0 +1,110 @@
+import { Logger, Messages } from '@salesforce/core';
+import { AsyncCreatable } from '@salesforce/kit';
+import { uxEvents } from '../ScannerEvents';
+
+
+Messages.importMessagesDirectory(__dirname);
+const messages = Messages.loadMessages('@salesforce/sfdx-scanner', 'EventKeyTemplates');
+
+export type PmdCatalogEvent = {
+    key: string;
+    args: string[];
+    type: string;
+    log: string;
+    handler: string;
+    verbose: boolean;
+    time: number;
+};
+
+/**
+ * Helps with processing output from PmdCatalog java module and converting messages into usable events
+ */
+export class OutputProcessor extends AsyncCreatable {
+
+    private logger!: Logger;
+    private messageLogger!: Logger;
+
+    protected async init(): Promise<void> {
+        this.logger = await Logger.child('OutputProcessor');
+        this.messageLogger = await Logger.child('MessageLog');
+    }
+
+    // We want to find any events that were dumped into stdout or stderr and turn them back into events that can be thrown.
+    // As per the convention outlined in SfdxMessager.java, SFDX-relevant messages will be stored in the outputs as JSONs
+    // sandwiched between 'SFDX-START' and 'SFDX-END'. So we'll find all instances of that.
+    public processOutput(stdout: string, stderr: string): void {
+        this.logger.trace(`stdout: ${stdout}`);
+        this.logger.trace(`stderr: ${stderr}`);
+
+        const outEvents: PmdCatalogEvent[] = this.getEventsFromString(stdout);
+        // TODO: After code streamlining, Main.java prints all messages at once to stdout. Consider removing stderr processing
```

Comment:
Question: Why did we change it so everything is logged in stdout, instead of having info/logs go into stdout and errors/warnings go in stderr? This seems fine too, I'm just curious what motivated the change.

Reply:
Having messages as items in a list would help with testing. Also, if at some point we decided to use a different mechanism to pass messages to another layer, having them as messages would give us a choice as to what we want to do with it. Converting to `List<Message>` lead to the change of printing all messages at once in the `finally` block of `main()`. At this point, the messages already have the information on how important they are (`MessageType`). So I went ahead with printing them all in `stdout`.